core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -49,7 +54,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     properties.remove(Properties.BatchReadableWritable.NAME);
     properties.remove(Properties.BatchReadableWritable.TYPE);
 
-    pipelineConfigurer.createDataset(datasetName, datasetType, DatasetProperties.builder().addAll(properties).build());
+    // null check for tests, as macro fields map and properties are null in test cases
```

Comment:
in tests, https://github.com/caskdata/hydrator-plugins/blob/develop/core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/TableSinkTest.java#L45

we create a TableSinkConfig manually, but since this is constructed using the constructor, the parent class fields `macroFields` and `properties` would be null. 

without this check, it will throw NPE,  we do a similar check here,
https://github.com/caskdata/hydrator-plugins/blob/develop/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java#L85